internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/endToEnd/PartialUpdateTest.java

```diff
@@ -1283,29 +1296,35 @@ public void testEnablePartialUpdateOnActiveActiveStore() {
           parentControllerClient,
           30,
           TimeUnit.SECONDS);
-      // Version 1 has active-active and write compute disabled, so each server host will ingest from local data center
-      // for leader.
-      verifyConsumerThreadPoolFor(
-          storeVersionTopicV1,
-          new PubSubTopicPartitionImpl(realTimeTopic, 0),
-          ConsumerPoolType.NON_CURRENT_VERSION_NON_AA_WC_LEADER_POOL,
-          1);
-      verifyConsumerThreadPoolFor(
-          storeVersionTopicV1,
-          new PubSubTopicPartitionImpl(storeVersionTopicV1, 0),
-          ConsumerPoolType.NON_CURRENT_VERSION_NON_AA_WC_LEADER_POOL,
-          1);
-      // Version 2 has active-active enabled, so each server host will ingest from two data centers for leader.
-      verifyConsumerThreadPoolFor(
-          storeVersionTopicV2,
-          new PubSubTopicPartitionImpl(realTimeTopic, 0),
-          ConsumerPoolType.CURRENT_VERSION_AA_WC_LEADER_POOL,
-          NUMBER_OF_CHILD_DATACENTERS);
-      verifyConsumerThreadPoolFor(
-          storeVersionTopicV2,
-          new PubSubTopicPartitionImpl(storeVersionTopicV2, 0),
-          ConsumerPoolType.CURRENT_VERSION_NON_AA_WC_LEADER_POOL,
-          1);
+      TestUtils.waitForNonDeterministicAssertion(ASSERTION_TIMEOUT_MS, TimeUnit.MILLISECONDS, true, () -> {
+        // Version 1 has active-active and write compute disabled, so each server host will ingest from local data
+        // center for leader.
+        verifyConsumerThreadPoolFor(
+            storeVersionTopicV1,
+            realTimeTopicPartition,
+            ConsumerPoolType.NON_CURRENT_VERSION_NON_AA_WC_LEADER_POOL,
+            1,
+            1);
+        verifyConsumerThreadPoolFor(
+            storeVersionTopicV1,
+            storeVersionTopicPartitionV1,
+            ConsumerPoolType.NON_CURRENT_VERSION_NON_AA_WC_LEADER_POOL,
+            1,
+            REPLICATION_FACTOR - 1);
+        // Version 2 has active-active enabled, so each server host will ingest from two data centers for leader.
+        verifyConsumerThreadPoolFor(
+            storeVersionTopicV2,
+            realTimeTopicPartition,
+            ConsumerPoolType.CURRENT_VERSION_AA_WC_LEADER_POOL,
+            NUMBER_OF_CHILD_DATACENTERS,
+            1);
+        verifyConsumerThreadPoolFor(
+            storeVersionTopicV2,
+            storeVersionTopicPartitionV2,
+            ConsumerPoolType.CURRENT_VERSION_NON_AA_WC_LEADER_POOL,
+            1,
+            REPLICATION_FACTOR - 1);
```

Comment:
Why expected replicas are one less than the RF?

Reply:
Discussed offline. A bit confusing here.

For hybrid store, leader is rt partition 1 per colo, follower are version topic partitions, it is RF-1.